python/requirements.txt

```diff
@@ -6,15 +6,16 @@ git+https://github.com/bystrogenomics/liftover.git@v1.1.18
 maturin[zig]==1.3.2
```

Comment:
Since I was adding dependencies, I took the opportunity to upgrade some of our key dependencies as well